src/Components/Web.JS/src/Virtualize.ts

```diff
@@ -0,0 +1,76 @@
+export const Virtualize = {
+  init,
+  dispose,
+};
+
+const observersByDotNetId = {};
+
+function findClosestScrollContainer(element: Element | null): Element | null {
+  if (!element) {
+    return null;
+  }
+
+  const style = getComputedStyle(element);
+
+  if (style.overflowY !== 'visible') {
+    return element;
+  }
+
+  return findClosestScrollContainer(element.parentElement);
+}
+
+function init(dotNetHelper: any, topSpacer: Element, bottomSpacer: Element, rootMargin = 50): void {
+  const intersectionObserver = new IntersectionObserver(intersectionCallback, {
+    root: findClosestScrollContainer(topSpacer),
+    rootMargin: `${rootMargin}px`,
+  });
+
+  intersectionObserver.observe(topSpacer);
+  intersectionObserver.observe(bottomSpacer);
+
+  const mutationObserver = new MutationObserver((): void => {
+    intersectionObserver.unobserve(topSpacer);
+    intersectionObserver.unobserve(bottomSpacer);
+    intersectionObserver.observe(topSpacer);
+    intersectionObserver.observe(bottomSpacer);
+  });
+
+  // Observe the bottom spacer to account for collections that resize.
+  mutationObserver.observe(bottomSpacer, { attributes: true });
+
+  observersByDotNetId[dotNetHelper._id] = {
```

Comment:
Minor point, but rather than relying on the internal `_id` value here, would it work to implement `observersByDotNetId` as a `Map<any, Virtualize>` so that the `dotNetHelper` instance itself could be the key?

Reply:
Unfortunately, it seemed that all instances of `dotNetHelper` were treated as being the same key since they are all stringified to `[object Object]`.

Reply:
Sorry I wasn't more explicit - I mean defining `observersByDotNetId` like:

```ts
const observersByDotNetId = new Map<any, Virtualize>();
```

... so it's an actual instance of `Map` (not a plain JS object). That way it can distinguish keys of arbitrary object types and won't stringify the keys at all.

If I'm missing something about how this works please let me know!

Reply:
Looks like I was using `Map` incorrectly by adding properties using `[]` instead of `.set` which is why I didn't observe the correct behavior.

Reply:
Update: Unfortunately, I think using `Map` still won't work - It looks like `dotNetHelper` instances don't have reference equality between calls on the JS side, so they aren't recognized as valid keys in the map (at least that's what I think is happening).

Reply:
Ah, makes sense. Thanks for checking this out and sorry for the false lead. Using `_id` is fine.